docker_mcp/services/stack.py

```
@@ -402,6 +423,36 @@ async def migrate_stack(
                 self._build_ssh_cmd(target_host), target_appdata, stack_name
             )
             
+            # Step 5.1: CRITICAL - Backup existing target data before ANY changes
+            backup_info = None
+            if not dry_run:
+                migration_steps.append(f"💾 Creating backup of existing target data...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)